otter/convergence/service.py

```diff
@@ -1,17 +1,87 @@
 """Converger service"""
 
+import time
+
 from functools import partial
 
 from effect import Effect
 from effect.twisted import perform
 
+from toolz.itertoolz import concat
+
 from twisted.application.service import Service
 
 from otter.constants import CONVERGENCE_LOCK_PATH
-from otter.convergence.composition import (
-    execute_convergence, get_desired_group_state)
+from otter.convergence.composition import get_desired_group_state
+from otter.convergence.effecting import steps_to_effect
+from otter.convergence.gathering import get_all_convergence_data
+from otter.convergence.model import ServerState
+from otter.convergence.planning import plan
 from otter.http import TenantScope
+from otter.models.intents import ModifyGroupState
 from otter.util.deferredutils import with_lock
+from otter.util.fp import assoc_obj
+
+
+def server_to_json(server):
+    """
+    Convert a NovaServer to a dict representation suitable for returning to the
+    end-user as a part of group state.
+    """
+    return {'id': server.id}
+
+
+def determine_active(servers, lb_nodes):
+    """
+    Given the current NovaServers and LB nodes, determine which servers are
+    completely built.
+
+    :param servers: sequence of :obj:`NovaServer`.
+    :param lb_nodes: sequence of :obj:`ILBNode`.
+
+    :return: list of servers that are active.
+    """
+
+    def all_met(server, current_lb_nodes):
+        """Determine if a server is in all the LBs it wants to be in."""
+        desired_lbs = set(concat(server.desired_lbs.values()))
```

Comment:
Non-blocking thought: Do you think it will be better to implement #1049 and change this?